### PR TITLE
Bug 1245170: Allow tooltool uploads with non-user-associated tokens

### PR DESCRIPTION
**Pull request by djmitche**
Still needs tests; see https://bugzilla.mozilla.org/show_bug.cgi?id=1245170 for the motivation.

### CODE REVIEW

File: relengapi/docs/relnotes/index.rst [hunk 1]
```diff
@@ -2,6 +2,7 @@ Release Notes
 =============
 
 .. toctree::
+    3.3.4
```
Comment (Callek) [inline relengapi/docs/relnotes/index.rst]: Looks like you never merged 3.3.4 into master, I don't see why it should be part of this PR...

**Comment by Callek** [comment]
I'm also confused on the relative spurious use of newlines in this PR, its as if select lines were autopep8'd without adjusting the --max-line-length. I'd personally rather we avoid the [seemingly] spurious changes.

File: relengapi/docs/relnotes/index.rst (hunk 1 above)
Comment (djmitche) [inline relengapi/docs/relnotes/index.rst]: Oops -- pushed to master.  @garbas this is why we should have CD :)

**Comment by djmitche** [comment]
Sorry about the line wraps -- I didn't mean to!

**Comment by Callek** [comment]
offhand I'm not sure if the test supports the Token auth type (as in, if we use a token we're not necessarily testing that tokenauth works here -- just a 'special' user auth, so that might be a room for testing improvement given the goal of this PR)

That said, I don't think blocking on it is worthwhile at all, so :shipit:

**Comment by garbas** [comment]
@djmitche code looks straightforward +1 on that, but I'm not sure I understand the whole picture to be able to judge how this affects other things.

File: relengapi/docs/relnotes/index.rst (hunk 1 above)
Comment (garbas) [inline relengapi/docs/relnotes/index.rst]: Quoting Dustin J. Mitchell (2016-02-03 16:11:41)
> In relengapi/docs/relnotes/index.rst:
> 
> > @@ -2,6 +2,7 @@ Release Notes
> >  =============
> >
> >  .. toctree::
> > +    3.3.4
> 
> Oops -- pushed to master. @garbas this is why we should have CD :)
> 

give me a week and i'll get it done! :)

**Pull request merged**